pyuvdata/uvdata/fhd.py

```diff
@@ -205,17 +226,22 @@ def read_fhd(
         if settings_file is None:
             warnings.warn("No settings file included in file list")
 
-        # TODO: add checking to make sure params, flags and datafiles are
-        # consistent with each other
-        vis_data = {}
-        for pol, file in datafiles.items():
-            this_dict = readsav(file, python_dict=True)
-            if use_model:
-                vis_data[pol] = this_dict["vis_model_ptr"]
-            else:
-                vis_data[pol] = this_dict["vis_ptr"]
-            this_obs = this_dict["obs"]
-            data_shape = vis_data[pol].shape
+        if not read_data:
+            obs_dict = readsav(obs_file, python_dict=True)
+            this_obs = obs_dict["obs"]
+            self.Npols = int(this_obs[0]["N_POL"])
+        else:
+            # TODO: add checking to make sure params, flags and datafiles are
+            # consistent with each other
+            vis_data = {}
+            for pol, file in datafiles.items():
+                this_dict = readsav(file, python_dict=True)
+                if use_model:
+                    vis_data[pol] = this_dict["vis_model_ptr"]
+                else:
+                    vis_data[pol] = this_dict["vis_ptr"]
+                this_obs = this_dict["obs"]
+            self.Npols = len(list(vis_data.keys()))
```

Comment:
Is it possible for this determination of `Npols` to be different than the one above? Should we do that consistency check?

Reply:
@bhazelton do you know the answer to this? I'm not familiar with FHD.

Reply:
It can be. For FHD, each polarization is written to a different file. If you provide data files, the number of polarizations will be the number of files you provided. If you do not provide the data files then you can't count it that way and you need to use the number in the `obs` structure, which tells you how many pols the FHD run that produced it used (which is an upper limit on the number you could pass it).

Reply:
I can add a comment to this effect in the docstring or in the code if you think I should.

Reply:
Thanks for the info. It might help to clarify that when we do a metadata-only read, we get the number of available pols (if I'm understanding correctly), whereas for actual data, it's related to the number of files passed in.

Reply:
ok, I just pushed a commit with a comment about this in the docstring.

Reply:
Looks great, thanks!